docs/migration-to-NRI.md

```diff
@@ -0,0 +1,113 @@
+
+- ### For containerd edit the containerd config file and look for the section "io.containerd.nri.v1.nri" and replace "disable = true" with "disable = false":
+  - ```console
+    vim /etc/containerd/config.toml
+    ```
+
+- ### For crio edit the crio config and look for the "[crio.nri]" section, replace "enable_nri = false" with "enable_nri = true" and uncomment the NRI config lines.
+  - ```console
+    sudo vim /etc/crio/crio.conf
+    ```
+
+- ### Build the policies:
+  - ```console
+    git clone https://github.com/jukkar/nri-resmgr.git
+  - ```console
+    cd nri-resmgr
+    ````
+  - Switch to a stable branch
+  - ```console
+    git checkout preview
+  - ```console
+    make
+  - ```console
+    make images
+    ````
+
+- ### Apply required crds:
+  - ```console
+    kubectl apply -f deployment/base/crds/noderesourcetopology_crd.yaml
+    ````
+
+- ### Import the image and start the NRI plugin you want to run:
+  - For Containerd:
+  - ```console
+    ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar
+  - For CRI-O:
+  - <b>See step 3 how to push the images to a registry, then pull from there.</b>
+  - Then deploy the plugin:
+  - ```console
+    kubectl apply -f build/images/nri-resmgr-topology-aware-deployment.yaml
+    ```
+
+- ### Deploy your test pod.
+
+- ### See the resources the pod got assigned with:
+  - ```console
+    kubectl exec $pod -c $container  -- grep allowed_list: /proc/self/status
+  - ```console
+    Output should look similar to the output of CRI-RM
+    ```
+
+## 3. Steps for a more real-life migration using self-hosted image repository.
+- ### Same steps as above for enabling NRI with Containerd/CRI-O and building the images.
+- ### Load the .tar file into your image repository:
+  - ```console
+    docker load -i build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar
+    ````
+
+- ### Tag the image with your registry:
+  - ```console
+    docker tag myimage:<version> myregistry.com/myimage:<version>
+    ```
+
+- ### Push the image to your registry:
+  - ```console
+    docker push myregistry.com/myimage:<version>
+    ```
+
+- ### Remember to change the image name & pull policy in the plugins .yaml file to match your registyr and image, ex:
+  - ```console
+    vim build/images/nri-resmgr-topology-aware-deployment.yaml
+    ```
+
+- ### Then deploy the plugin simlarly to the earlier step.
+
+## 4. Migrating existing configuration
+- The ConfigMap used by the ported policies/infra has a different name/naming scheme than the original one used in CRI-RM, ex:
+  - configMapName: "configmap-name: cri-resmgr-config" is now "configmap-name: nri-resmgr-config"
+
+## 5. Migrating existing workloads
+- The annotations one can use to customize how a policy treats a workload use slightly different keys than the original ones in CRI-RM. ex:
+  - ResmgrKeyNamespace: ResmgrKeyNamespace = "cri-resource-manager.intel.com" is now ResmgrKeyNamespace = "nri-resmgr.intel.com"
```

Comment:
Maybe we could state this slightly differently. I suspect we can't expect folks to make the right mental association based on `ResmgKeyNamespace` unless they are familiar with the cri-resmgr source code. So maybe something along the lines of

The annotations one can use to customize how a policy treats a workload use slightly different keys than the original ones in CRI-RM. The collective 'key namespace' for policy- and resource-manager-specific annotation has been changed from ` cri-resource-manager.intel.com` to `nri-resmgr.intel.com`.

For instance, an explicit type annotation for the `balloons` policy, which used to be
```
...
metadata:
  annotations:
    balloon.balloons.cri-resource-manager.intel.com/container.$CONTAINER_NAME: $BALLOON_TYPE`
...
```

should now be  

```
...
metadata:
  annotations:
    balloon.balloons.nri-resmgr.intel.com/container.$CONTAINER_NAME: $BALLOON_TYPE`
...
```

Similarly a workload opt-out annotation from exclusive CPU allocation for the `topology-aware` policy, which used to be

```
...
metadata:
  annotations:
    prefer-shared-cpus.cri-resource-manager.intel.com/container.$CONTAINER_NAME: "true"
...
```

should now be 

```
...
metadata:
  annotations:
    prefer-shared-cpus.nri-resmgr.intel.com/container.$CONTAINER_NAME: "true"
...
```

Similar changes are needed for any cri-resmgr-specific annotation that uses the same semantic scoping for key syntax.